docs/core/howto/defer-intro.rst

```diff
@@ -397,24 +401,21 @@ Coroutines with async/await
 
 .. note::
 
-    Only available on Python 3.5 and higher.
-
     .. versionadded:: 16.4
 
-On Python 3.5 and higher, the :pep:`492` ("Coroutines with async and await syntax") "await" functionality can be used with Deferreds by the use of :api:`twisted.internet.defer.ensureDeferred <ensureDeferred>`.
-It is similar to ``inlineCallbacks``, except that it uses the ``await`` keyword instead of ``yield``, the ``return`` keyword instead of ``returnValue``, and is a function rather than a decorator.
-
-Calling a coroutine (that is, the result of a function defined by ``async def funcname():``) with :api:`twisted.internet.defer.ensureDeferred <ensureDeferred>` will allow you to "await" on Deferreds inside it, and will return a standard Deferred.
-You can mix and match code which uses regular Deferreds, ``inlineCallbacks``, and ``ensureDeferred`` freely.
+Python 3.5 introduced :pep:`492` ("Coroutines with async and await syntax") and native coroutines.
+:api:`twisted.internet.defer.Deferred.fromCoroutine <Deferred.fromCoroutine>` allows you to write coroutines with the ``async def`` syntax and ``await`` on Deferreds, similar to ``inlineCallbacks``.
```

Comment:
If someone is reading this document in order to *understand* some code that's calling `ensureDeferred`, the lack of any mention here might be a little confusing.  I think it might be good to put in an aside here, documenting `ensureDeferred` and saying "earlier versions of Twisted only had the more type-ambiguous `ensureDeferred`..." or something, at least until we deprecate it.

Reply:
@glyph how's this: https://github.com/twisted/twisted/pull/1269/files#diff-f6e9f4618d9eb1551a84a53bd2700661R329